packages/features/ee/round-robin/roundRobinReassignment.ts

```diff
@@ -0,0 +1,365 @@
+    uid: booking.uid,
+    destinationCalendar,
+    team: {
+      members: teamMembers,
+      name: eventType.team?.name || "",
+      id: eventType.team?.id || 0,
+    },
+    customInputs: isPrismaObjOrUndefined(booking.customInputs),
+    ...getCalEventResponses({
+      bookingFields: eventType?.bookingFields ?? null,
+      booking,
+    }),
+  };
+
+  const credentials = await prisma.credential.findMany({
+    where: {
+      userId: organizer.id,
+    },
+    include: {
+      user: {
+        select: {
+          email: true,
+        },
+      },
+    },
+  });
+  const eventManager = new EventManager({ ...organizer, credentials: [...credentials] });
+
+  await eventManager.reschedule(
+    evt,
+    booking.uid,
+    undefined,
+    hasOrganizerChanged,
+    previousHostDestinationCalendar ? [previousHostDestinationCalendar] : []
+  );
+
+  // Send to new RR host
+  await sendRoundRobinScheduledEmails(evt, [
+    {
+      ...reassignedRRHost,
+      name: reassignedRRHost.name || "",
+      username: reassignedRRHost.username || "",
+      timeFormat: getTimeFormatStringFromUserTimeFormat(reassignedRRHost.timeFormat),
+      language: { translate: reassignedRRHostT, locale: reassignedRRHost.locale || "en" },
+    },
+  ]);
+  // Send to cancelled RR host
+  await sendRoundRobinCancelledEmails(evt, [
+    {
+      ...previousRRHost,
+      name: previousRRHost.name || "",
+      username: previousRRHost.username || "",
+      timeFormat: getTimeFormatStringFromUserTimeFormat(previousRRHost.timeFormat),
+      language: { translate: previousRRHostT, locale: previousRRHost.locale || "en" },
+    },
+  ]);
+
+  // Handle only email host workflows
+  const workflows = await prisma.workflow.findMany({
+    where: {
+      OR: [
+        {
+          trigger: WorkflowTriggerEvents.NEW_EVENT,
+        },
+        {
+          trigger: WorkflowTriggerEvents.BEFORE_EVENT,
+        },
+      ],
+      activeOn: {
+        some: {
+          eventTypeId: eventType.id,
+        },
+      },
+      steps: {
+        some: {
+          action: WorkflowActions.EMAIL_HOST,
+        },
+      },
+    },
+    include: {
+      steps: true,
+    },
+  });
+
+  for (const workflow of workflows) {
+    // Only trigger new host step
+    workflow.steps = workflow.steps.filter((step) => step.action === WorkflowActions.EMAIL_HOST);
+
+    for (const step of workflow.steps) {
+      await scheduleEmailReminder({
+        evt: {
+          ...evt,
+          eventType,
+        },
+        action: WorkflowActions.EMAIL_HOST,
+        triggerEvent: workflow.trigger,
+        timeSpan: {
+          time: workflow.time,
+          timeUnit: workflow.timeUnit,
+        },
+        sendTo: reassignedRRHost.email,
+        template: step.template,
+      });
+    }
+  }
+
+  // TODO create relationship between reminder and userId in order to determine which one to delete
+  // deleteScheduledEmailReminder(reminder.id, reminder.referenceId);
+};
```

Comment:
@CarinaWolli I was thinking about this. Is this something we have to do or am I over thinking it?